st2common/st2common/util/actionalias_matching.py

```diff
@@ -0,0 +1,90 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import six
+
+from st2common.exceptions.content import ParseException
+from st2common.models.utils.action_alias_utils import extract_parameters
+
+__all__ = [
+    'list_format_strings_from_aliases',
+    'normalise_alias_format_string',
+    'match_command_to_alias'
+]
+
+
+def list_format_strings_from_aliases(aliases):
+    '''
+    List patterns from a collection of alias objects
+
+    :param aliases: The list of aliases
+    :type  aliases: ``list`` of :class:`st2common.models.api.action.ActionAliasAPI`
+
+    :return: A description of potential execution patterns in a list of aliases.
+    :rtype: ``list`` of ``list``
+    '''
+    patterns = []
+    for alias in aliases:
+        for _format in alias.formats:
```

Comment:
`format_` (yeah, blame pep8 - https://www.python.org/dev/peps/pep-0008/#function-and-method-arguments :D)